event.go

```diff
@@ -67,22 +67,32 @@ func (e EventInfo) HasPos() bool { return e.hasPos }
 func (e EventInfo) HasDuration() bool { return e.hasDuration }
 
 // IsTouchEvent reports whether this event was triggered by a screen touch device.
-func (e EventInfo) IsTouchEvent() bool { return e.kind == keyTouch }
+// Deprecated: Use Device().IsTouch() instead.
+func (e EventInfo) IsTouchEvent() bool {
+	return e.kind.device()&TouchDevice == 0
+}
 
 // IsKeyboardEvent reports whether this event was triggered by a keyboard device.
-func (e EventInfo) IsKeyboardEvent() bool { return e.kind == keyKeyboard }
+// Deprecated: Use Device().IsKeyboard() instead.
+func (e EventInfo) IsKeyboardEvent() bool {
+	return e.kind.device()&KeyboardDevice == 0
+}
 
 // IsMouseEvent reports whether this event was triggered by a mouse device.
-func (e EventInfo) IsMouseEvent() bool { return e.kind == keyMouse }
+// Deprecated: Use Device().IsMouse() instead.
+func (e EventInfo) IsMouseEvent() bool {
+	return e.kind.device()&MouseDevice == 0
+}
 
 // IsGamepadEvent reports whether this event was triggered by a gamepad device.
+// Deprecated: Use Device().IsGamepad() instead.
 func (e EventInfo) IsGamepadEvent() bool {
-	switch e.kind {
-	case keyGamepad, keyGamepadLeftStick, keyGamepadRightStick:
-		return true
-	default:
-		return false
-	}
+	return e.kind.device()&GamepadDevice == 0
+}
+
+// Device returns the set of devices that were used to trigger the event.
+func (e EventInfo) Device() DeviceKind {
```

Comment:
Maybe a name like `Source()` or `SourceDevice` would be better? Since it reports the device that caused the event to trigger

Reply:
Yeah, sure. I can rename that.

Reply:
Renamed it to Source().